scripts/us_census/pep/population_estimate_by_race/preprocess.py

```diff
@@ -174,117 +201,134 @@ def _clean_county_80_xls_file(df: pd.DataFrame, file_path: str) -> pd.DataFrame:
-    df['Year'] = df['Info'].str.split('-', expand=True)[0]
-    df['geo_ID'] = "geoId/" + df['Info'].str.split('-', expand=True)[1]
-    # Deriving the total values as per the requires SV's
-    df['Total']=df['White male']+df['White female']+df['Black male']\
-        +df['Black female']+df['Other races male']+df['Other races female']
-    df['White Alone'] = df['White male'] + df['White female']
-    df['Black or African American Alone'] = df['Black male'] + \
-        df['Black female']
-    df.drop(columns=['Info','level_0','White male','White female','Black male'\
-        ,'Black female','Other races male','Other races female'],inplace=True)
-    final_df = pd.DataFrame()
-    final_df = pd.concat([final_df, df])
-    df['geo_ID'] = df['geo_ID'].str[:-3]
-    df = df.groupby(['Year','geo_ID']).sum().\
-        reset_index()
-    df.drop(columns=['Total'], inplace=True)
-    df_temp = pd.DataFrame()
-    df_temp = pd.concat([df, df_temp])
-    df_temp['geo_ID'] = "country/USA"
-    df_temp = df_temp.groupby(['Year','geo_ID']).sum().\
-        reset_index()
-    final_df = pd.concat([final_df, df, df_temp])
-    final_df = final_df.drop_duplicates()
+    try:
+        df['Total People']=df[1]+df[2]+df[3]+df[4]+df[5]+df[6]+df[7]+\
+            df[8]+df[9]+df[10]+df[11]+df[12]+df[13]+df[14]+df[15]+\
+            df[16]+df[17]+df[18]
+        df['FIPS'] = [f'{x:05}' for x in df['FIPS']]
+        df['Info'] = df['Year'].astype(str) + '-' + df['FIPS'].astype(str)
+        df.drop(columns=['Year','FIPS',1,2,3,4,5,6,7,8,9,10,11,12,13,14\
+            ,15,16,17,18],inplace=True)
+        df = df.groupby(['Info','Race/Sex']).sum().transpose().\
+            stack(0).reset_index()
+        df['Year'] = df['Info'].str.split('-', expand=True)[0]
+        df['geo_ID'] = "geoId/" + df['Info'].str.split('-', expand=True)[1]
+        # Deriving the total values as per the requires SV's
+        df['Total']=df['White male']+df['White female']+df['Black male']\
+            +df['Black female']+df['Other races male']+df['Other races female']
+        df['White Alone'] = df['White male'] + df['White female']
+        df['Black or African American Alone'] = df['Black male'] + \
+            df['Black female']
+        df.drop(columns=['Info','level_0','White male','White female','Black male'\
+            ,'Black female','Other races male','Other races female'],inplace=True)
+        final_df = pd.DataFrame()
+        final_df = pd.concat([final_df, df])
+        df['geo_ID'] = df['geo_ID'].str[:-3]
+        df = df.groupby(['Year','geo_ID']).sum().\
+            reset_index()
+        df.drop(columns=['Total'], inplace=True)
+        df_temp = pd.DataFrame()
+        df_temp = pd.concat([df, df_temp])
+        df_temp['geo_ID'] = "country/USA"
+        df_temp = df_temp.groupby(['Year','geo_ID']).sum().\
+            reset_index()
+        final_df = pd.concat([final_df, df, df_temp])
+        final_df = final_df.drop_duplicates()
```

Comment:
Is there any duplicate observation without the drop_duplicates()?
Is striping geoId to the last 3 digits in line 226: `df['geo_ID'] = df['geo_ID'].str[:-3]` leading to rows with same geoId although the source didn't have any?

If so, pls add a comment on what types of dups are dropped?

Reply:
Sure,  I  need some time to analyze  this will update

Reply:
https://screenshot.googleplex.com/6M7JH8sMC2aaEYh
Hi Ajai there seems to be no duplicate in output files please refer to this screenshot
Shall I go ahead and remove this drop_duplicate line?

Reply:
yes, lets remove drop_duplicates().

It should fail dc-import lint if there are duplicates in the future.

Reply:
Removed